src/test/java/com/google/sps/ImageSelectionTest.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.image;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+@RunWith(JUnit4.class)
+public final class ImageSelectionTest {
+
+  private Set<String> keywords;
+
+  private static final String bingImgLinkStart = "https://tse";
+
+  private static void checkLink(ImageSelection imageSelection) {
+    try {
+      String actualLink = imageSelection.getBestImage();
+      Assert.assertEquals(bingImgLinkStart, actualLink.substring(0, 11));
+    } catch(IOException ex) {
+      // Test fail because the function threw an exception
+      Assert.fail(ex.getMessage());
+    }
+  }
+
+  @Before
+  public void setUp() {
+    keywords = new LinkedHashSet<>();
+  }
+
+  @Test
+  public void fewKeywordsUrlGenerator() {
+    keywords.add("london");
+
+    ImageSelection imageSelection = new ImageSelection(keywords);
+    checkLink(imageSelection);
+  }
+
+  @Test
+  public void moreKeywordsUrlGenerator() {
```

Comment:
I recommend to have the test name include a hint at what behavior the test checks.

e.g. you could say `multipleKeywordsXyz` where Xyz hints at what the desired output is.

This comment applies to other methods here as well.